lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -940,49 +940,42 @@ def subsequent_wallet_auth(payment_method, options)
       end
 
       def add_auth_network_tokenization(xml, payment_method, options)
-        brand = card_brand(payment_method)
+        brand = card_brand(payment_method)&.to_sym
         return if payment_method.mobile_wallet? && brand == 'discover' && !options[:enable_cybs_discover_apple_pay]
 
         if payment_method.network_token?
           xml.tag!('networkTokenCryptogram', payment_method.payment_cryptogram)
           xml.tag!('commerceIndicator', stored_credential_commerce_indicator(options) || 'internet')
-        elsif send_only_commerce_indicator(payment_method, options)
-          commerce_indicator = if brand == 'discover'
+        elsif send_only_commerce_indicator(brand, payment_method, options)
+          commerce_indicator = if brand == :discover
                                  'dipb'
                                elsif subsequent_wallet_auth(payment_method, options)
                                  'internet'
                                else
-                                 ECI_BRAND_MAPPING[brand.to_sym]
+                                 ECI_BRAND_MAPPING[brand]
                                end
 
           xml.commerceIndicator(commerce_indicator)
         else
-          default_wallet_values(xml, payment_method)
+          default_wallet_values(xml, brand, payment_method)
         end
       end
 
-      def send_only_commerce_indicator(payment_method, options)
-        brand = card_brand(payment_method)
+      def send_only_commerce_indicator(brand, payment_method, options)
+        return false if brand == :american_express
 
-        return false if brand == 'american_express'
-
-        brand == 'master' || subsequent_wallet_auth(payment_method, options)
+        brand == :master || subsequent_wallet_auth(payment_method, options)
       end
 
-      def default_wallet_values(xml, payment_method)
-        brand = card_brand(payment_method).to_sym
+      def default_wallet_values(xml, brand, payment_method)
         commerce_indicator = brand == :discover ? 'dipb' : ECI_BRAND_MAPPING[brand]
-        cryptogram = brand == :american_express ? Base64.decode64(payment_method.payment_cryptogram) : payment_method.payment_cryptogram
-        cavv = xid = cryptogram
 
-        if brand == :american_express
-          cavv = Base64.encode64(cavv[0...20])
-          xid = xid.bytes.count > 20 ? Base64.encode64(xid[20...40]) : nil
-        end
+        network_token_cryptogram = brand == :american_express ? Base64.decode64(payment_method.payment_cryptogram) : payment_method.payment_cryptogram
```

Comment:
There's no reason to decode the AmericanExpress cryptogram. Let's treat it like every other cryptogram. Cybersource said that the SOAP API integration for this should be the same as the REST one.